react/week3/extending-todo-app/src/App.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import List from './List';
+import './App.css';
+
+function App() {
+  const [description,setDescription]=useState("");
+  const [date,setDate]=useState("");
+  const [list, setList] = useState([]);
+
+  useEffect(() => {
+    fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw")
+      .then(res => res.json())
+      .then(
+        (result) => {
+          setList(result);
+        },
+        (error) => {
+          console.log(error)
+        }
+      )
+  }, [])
+
+
+
+  function editListItem(id,newDescription){
+    const newList=[...list];
+    const objIndex = newList.findIndex((obj => obj.id === id));
+    newList[objIndex].description=newDescription;
+    setList(newList);
```

Comment:
The only issue here is that on line 28 the original list object gets modified.
I know now you're thinking "but I copied the `list` array".

However.. the spread syntax is a little tricky in JavaScript, when you write `[...list]`, it will create a so called shallow copy, where only the first level is copied, any deeper levels will point to the original array/object. 
Eg. `newList[0] = "something new"` will not modify the original list array,
but(!) `newList[0].description = "something new"` will do it because this is one level deeper.

The way I would write this without modifying the `list` array:
```
const newList = list.map((item) => {
    if(item.id === id) {
        return {...item, description: newDescription}
    }

    return item
})
```
Here I create a new array with the `map` method (similary what the spread operator does). 
`map` will copy every `item` as they are, except the one I want to modify, for that one I copy `item` object using the spread syntax, and modify the `description` value only in the copied object.

`{...item, description: newDescription}` is just a shorthand for eg.:
```
const copiedItem = {...item};
copiedItem.description = newDescription;
return copiedItem
```